File: backend/Modules/Appointments/Modules.Appointments.Features/GetProfessionalAppointments/GetProfessionalAppointmentsDto.cs
using Modules.Appointments.Domain.Enums;
using Modules.Patients.PublicApi.Contracts;

namespace Modules.Appointments.Features.GetProfessionalAppointments;

public record GetProfessionalAppointmentsDto( 
    Guid Id,
    Guid PatientId,
    Guid ProfessionalId,
    string? Notes,
    DateTime StartDate,
    DateTime EndDate,
    AppointmentUrgency Urgency,
    AppointmentStatus Status,
    decimal Price,
    DateTime? OfferedAt,
    DateTime? ConfirmedAt,
    DateTime? CompletedAt,
    DateTime? CancelledAt,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    PatientDto Patient);

File: frontend/src/features/professional/hooks/GetProfessionalAppointments/GetProfessionalAppointmentsDto.ts
import type {
  AppointmentStatus,
  AppointmentUrgency,
  PatientDto,
} from "@/features/patient";

export interface GetProfessionalAppointmentsDto {
  id: string;
  patientId: string;
  professionalId: string;
  notes?: string;
  startDate: string;
  endDate: string;
  urgency: AppointmentUrgency;
  status: AppointmentStatus;
  price: number;
  offeredAt: string;
  confirmedAt: string;
  completedAt: string;
  cancelledAt: string;
  createdAt: string;
  updatedAt: string;
  patient: PatientDto;
}
